### ISSUE TITLE
Same host may not be able to load and use enclaves created from different EDLs

### ISSUE THREAD

**Issue by anakrish** [openenclave/openenclave#3250]
In v0.10, the SDK is built with `COMPILE_SYSTEM_EDL=off`. This means that system edls are not treated in a special manner and therefore the user has to explicitly import them in their own EDLs.

While this is a good direction (the user can control the ocalls made by the enclaves), currently this introduces a regression in v0.10.

Previously (inv v0.9 and earlier), a user can write a host application that loads and uses enclaves created from these two different edls.

```
enclave_a.edl:
enclave {
   trusted { 
       void a_func();
   };
};

enclave_b.edl:
enclave {
   trusted { 
       void b_func();
   };
};
```

enclave_a and enclave_b have non conflicting names and therefore the host can load and execute both the enclaves.

In v0.10, the user has to explicitly import the system EDLs:

```
enclave_a.edl:
enclave {
   from "openenclave/sgx/platform.edl" import *;
   trusted { 
       void a_func();
   };
};

enclave_b.edl:
enclave {
    from "openenclave/sgx/switchless.edl" import *;
    from "openenclave/sgx/platform.edl" import *;
 
   trusted { 
       void b_func();
   };
};
```
If the system EDLs are imported differently (as is the case above) then the ECALLs may end up with different ids for two enclaves, causing in runtime ecall dispatching failures.

PR #3086 describes the solution for this scenario. The design was reviewed and accepted. The implementation however did not make it to v0.10.

CC: @radhikaj @mingweishih

**Comment by radhikaj** [comment]
A work around for this is to create a single edl and stub out appropriate calls for each enclave.

**Comment by mingweishih** [comment]
Addressed by #3366.
